### PR TITLE
Add kube-apiserver-extra-sans option

### PR DESCRIPTION
**Pull request by petrutlucian94**
We're adding a config option that allows specifying additional kube-apiserver Subject Alternative Names.

Applicable spec: <link>

### Checklist

- [ ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [ ] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [ ] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: charms/worker/k8s/charmcraft.yaml [hunk 1]
```diff
@@ -224,6 +224,17 @@ config:
           runtime-config=batch/v2alpha1=true profiling=true
         will result in kube-apiserver being run with the following options:
           --runtime-config=batch/v2alpha1=true --profiling=true
+    kube-apiserver-extra-sans:
+      type: string
+      default: ""
+      description: |
+        Comma-separated list of extra sans for the kube-apiserver self-signed
+        certificates.
```
Comment (addyess) [inline charms/worker/k8s/charmcraft.yaml:232]: Would you accept space separated?

```suggestion
        Space-separated list of extra Subject Alternative Names for the kube-apiserver self-signed
        certificate.
```

File: charms/worker/k8s/src/charm.py [hunk 2]
```diff
@@ -317,6 +317,12 @@ def _check_k8sd_ready(self):
         status.add(ops.MaintenanceStatus("Ensuring snap readiness"))
         self.api_manager.check_k8sd_ready()
 
+    def _get_extra_sans(self):
+        """Retrieve the certificate extra SANs."""
+        extra_sans_str = str(self.config.get("kube-apiserver-extra-sans") or "")
+        configured_sans = [san.strip(" ") for san in extra_sans_str.split(",") if san.strip(" ")]
+        return [_get_public_address()] + configured_sans
```
Comment (addyess) [inline charms/worker/k8s/src/charm.py:324]: prevent duplicate sans being configured

```suggestion
        all_sans = set(extra_sans_str.strip().split()) | {_get_public_address()}
        return sorted(all_sans)
```

**Review by addyess — CHANGES_REQUESTED** [review]
Can we use space separated rather than comma separated?

File: charms/worker/k8s/charmcraft.yaml (hunk 1 above)
Comment (petrutlucian94) [reply charms/worker/k8s/charmcraft.yaml:232]: Sure. I see that some options expect comma separated values (``bootstrap-pod-cidr``) while others use space separation (``dns-upstream-nameservers``).

File: charms/worker/k8s/src/charm.py (hunk 2 above)
Comment (petrutlucian94) [reply charms/worker/k8s/src/charm.py:324]: Good point.

**Review by addyess — APPROVED** [review]
LGTM!  Nice

**Pull request merged**